learning/views.py

```diff
@@ -23,6 +23,15 @@ def display_character(request, character_pk):
     except ObjectDoesNotExist:
         character = Character.objects.filter(pk__gt=character_pk).first()
         return redirect('display_character', character_pk=character.pk)
+    try:
+        status = request.session['is_learning']
+    except KeyError:
+        status = False
+    if status == True:
+        learning_process(request)
+    else:
+        request.user.last_study_vocab_count = 0
```

Comment:
After changing a model you would have to call its save methods to write it into DB,
here it would be `request.user.save()`

Reply:
also why do we need to reset `last_study_vocab_count` if user is not learning? User may be just viewing the character. The character display view/template are also used later for dictionary function.